spark_client/domain.py

```diff
@@ -0,0 +1,252 @@
+import io
+import os
+import re
+from dataclasses import dataclass
+from typing import Any, Callable, Dict, List, Optional
+
+from spark_client.utils import WithLogging, union
+
+
+class PropertyFile(WithLogging):
+    """Class for providing basic functionalities for IO properties files."""
+
+    def __init__(self, props: Dict[str, Any]):
+        """Initialize a PropertyFile class with data provided by a dictionary.
+
+        Args:
+            props: input dictionary
+        """
+        self.props = props
+
+    def __len__(self):
+        """Return the size of the property dictionary, i.e. the number of configuration parameters."""
+        return len(self.props)
+
+    @classmethod
+    def _is_property_with_options(cls, key: str) -> bool:
+        """Check if a given property is known to be options-like requiring special parsing.
+
+        Args:
+            key: Property for which special options-like parsing decision has to be taken
+        """
+        return key in ["spark.driver.extraJavaOptions"]
+
+    @classmethod
+    def _read_property_file_unsafe(cls, name: str) -> Dict:
+        """Read properties in given file into a dictionary.
+
+        Args:
+            name: file name to be read
+        """
+        defaults = dict()
+        with open(name) as f:
+            for line in f:
+                kv = list(filter(None, re.split("=| ", line.strip())))
```

Comment:
I don't like variables with names like `k` and `kv` and `kv2`. Should be explicitly named, for example `key` or `key_value` IMO. Makes the code harder to read otherwise.

Reply:
yes. Why not! I would address this!